intro-HPC/ch_MATLAB.tex

```diff
@@ -0,0 +1,117 @@
+flag is used (the \verb|-v| flag means verbose ouput).
+To show how \verb|mcc| can be used, we use the \verb|magicsquare| example
+that comes with MATLAB.
+
+\begin{prompt}
+%\shellcmd{cp \$EBROOTMATLAB/extern/examples/compiler/magicsquare.m .}%
+%\shellcmd{mcc -mv magicsquare.m}%
+Opening log file:  %\homedir/java.log.34090%
+Compiler version: 6.6 (R2018a)
+Dependency analysis by REQUIREMENTS.
+Parsing file "%\homedir/magicsquare.m%"
+	(Referenced from: "Compiler Command Line").
+Deleting 0 temporary MEX authorization files.
+Generating file "%\homedir/readme.txt%".
+Generating file "run\_magicsquare.sh".
+\end{prompt}
+
+\subsection{Libraries}
+
+To \emph{compile} a MATLAB library, the \verb|-l| flag should be used.
+
+To compile a MATLAB program that \emph{needs a library}, you can use the
+\texttt{-I \emph{library\_path}} flag. This will tell the compiler to also
+look for files in \texttt{\emph{library\_path}}.
+
+It's also possible to use the \texttt{-a \emph{path}} flag. That will result in
+all files under the \texttt{\emph{path}} getting added to the final executable.
+
+\subsection{Memory issues during compilation}
+
+If you are seeing Java memory issues during the compilation of your MATLAB program
+on the login nodes, consider tweaking the default maximum heap size (128M) of Java
+using the \verb|_JAVA_OPTIONS| environment variable with:
+
+\begin{prompt}
+%\shellcmd{export \_JAVA\_OPTIONS="-Xmx64M"}%
+\end{prompt}
+
+The MATLAB compiler spawns multiple Java processes, and because of the default memory
+limits that are in effect on the login nodes, this might lead to a crash of the compiler
+if it's trying to create to many Java processes. If we lower the heap size, more
+Java processes will be able to fit in memory.
+
+\section{Multithreading}
+
+MATLAB can only use the cores in a single
+workernode (unless the Distributed Computing toolbox is used, see
+\url{https://nl.mathworks.com/products/distriben.html}).
+
+The amount of workers used by MATLAB for the parallel toolbox can be controlled
+via the \verb|parpool| function: \verb|parpool('cluster1', 16)| will run the
+\verb|cluster1| profile with 16 workers.
+
+\section{Java output logs}
+
+Each time MATLAB is executed, it generates a Java log file in the users home directory.
+The output log directory can be changed using:
+
+\begin{prompt}
+%\shellcmd{MATLAB\_LOG\_DIR=\emph{<OUTPUT\_DIR>}}%
+\end{prompt}
+
+where \verb|<OUTPUT_DIR>| is the name of the desired output directory. To create
+and use a temporary directory for these logs:
+
+\begin{prompt}
+# create unique temporary directory in $TMPDIR (or /tmp/$USER if $TMPDIR is not defined)
+# instruct MATLAB to use this directory for log files by setting $MATLAB_LOG_DIR
+%\shellcmd{export MATLAB\_LOG\_DIR=\$(mktemp -d -p  \${TMPDIR:-/tmp/\$USER})}%
+\end{prompt}
+
+You should remove the directory at the end of your job script:
+\begin{prompt}
+%\shellcmd{rm -rf \$MATLAB\_LOG\_DIR}%
+\end{prompt}
+
+\section{MATLAB job script}
```

Comment:
Follow approach taken in https://github.com/hpcugent/vsc_user_docs/pull/149/files#diff-0a9de2ecd3a553e76e0e4c4246df3ad1R431